galley/pkg/runtime/monitoring/monitoring.go

```diff
@@ -134,6 +191,35 @@ func newView(measure stats.Measure, keys []tag.Key, aggregation *view.Aggregatio
 	}
 }
 
+func getStateTypeConfigKeys() ([]tag.Key, error) {
+	var err error
+	if NamespaceTag, err = tag.NewKey(namespace); err != nil {
+		panic(err)
+	}
+	if NameTag, err = tag.NewKey(name); err != nil {
+		panic(err)
+	}
+	if VersionTag, err = tag.NewKey(version); err != nil {
+		panic(err)
+	}
+
+	return []tag.Key{NamespaceTag, NameTag, VersionTag}, err
+}
+
+func registerNewStateTypeConfigView(collection string) error {
+	stateTypeConfigTotal[collection] = stats.Int64(fmt.Sprintf("galley/runtime/state/%s_total", collection),
```

Comment:
what does this leave the metric names as?  I feel like, at a minimum, we should we drop `state` and the `istio` bits that are in the middle here.

Reply:
name is something like: galley_runtime_state_istio_networking_destinationrules_total...
I like state in the name, istio certainly looks extra as its coming from galley which is part of istio
@ozevren : what do you think?

Reply:
I am ok, we leave only one prefix word (i.e. galley).
Galley can potentially serve some K8s resources (or other CRDs in general) as well. If we limit these metrics to Istio only, then we can drop Istio prefix (we should add the necessary logic in the name parsing code to handle that, if we go in that direction). but I'd rather keep it if possible. 

// istio is a good enough prefix to identify that the rest of the name is about config state.
galley_istio_networking_destinationrules?

Reply:
@douglas-reid any thoughts on this? I think this is the main sticking point that needs to be answered before we both can lgtm this PR.

Reply:
`galley_istio_networking_destinationrules` sgtm.

Reply:
Thanks! Changed to of the format: `galley_istio_networking_destinationrules`